boards/riscv/opentitan_earlgrey/doc/index.rst

```diff
@@ -0,0 +1,100 @@
+Github`_ page contains HDL code, utilities, and documentation relevant to the
+chip.
+
+Hardware
+********
+
+- RV32IMCB RISC-V "Ibex" core
+- 128kB main SRAM
+- Fixed-frequency and AON timers
+- 32 x GPIO
+- 4 x UART
+- 3 x I2C
+- 2 x SPI
+- Various security peripherals
+
+Detailed specification is on the `OpenTitan Earl Grey Chip Datasheet`_.
+
+Supported Features
+==================
+
+The ``opentitan_earlgrey`` board configuration is designed and tested to run on
+the Earl Grey chip simulated in Verilator, a cycle-accurate HDL simulation tool.
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| Timer     | on-chip    | RISC-V Machine Timer                |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling                 |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on Zephyr porting.
+
+Programming and Debugging
+*************************
+
+First, build and install Verilator as described in the `OpenTitan Verilator
+Setup`_ guide .
+
+Building and Flashing
+=====================
+
+Here is an example for building the :ref:`dining-philosophers-sample` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/philosophers
+   :board: opentitan_earlgrey
+   :goals: build
+
+The OpenTitan Vchip_sim_tb tool can take the Zephyr .elf as input and place it
+in simulated flash. The OpenTitan test ROM will then run in simulation, read
+the manifest header from simulated flash, and begin executing Zephyr from the
+entry point.
+
+.. code-block:: console
+
+   $OT_HOME/bazel-bin/hw/build.verilator_real/sim-verilator/Vchip_sim_tb --verbose-mem-load \
+   -r $OT_HOME/bazel-out/k8-fastbuild-ST-2cc462681f62/bin/sw/device/lib/testing/test_rom/test_rom_sim_verilator.39.scr.vmem \
+   --meminit=otp,$OT_HOME/bazel-out/k8-fastbuild/bin/hw/ip/otp_ctrl/data/img_rma.24.vmem \
+   --meminit=flash,$ZEPHYR_PATH/build/zephyr/zephyr.elf
+
+UART output:
+
+.. code-block:: console
+
+   I00001 test_rom.c:136] Version: earlgrey_silver_release_v5-8485-ga1284fc01, Build Date: 2022-11-10 13:45:25
+   I00002 test_rom.c:238] Test ROM complete, jumping to flash (addr: 20000384)!
+   *** Booting Zephyr OS build zephyr-v3.2.0-2723-g05fde9dafdc2 ***
+  Demo Description
+  ----------------
+  An implementation of a solution to the Dining Philosophers
+  problem (a classic multi-thread synchronization problem).
+  This particular implementation demonstrates the usage of multiple
+  preemptible and cooperative threads of differing priorities, as
+  well as dynamic mutexes and thread sleeping.
+  Philosopher 4 [C:-1]        STARVING
+  ...
```

Comment:
any reason for such an amount of details here? Maybe just hello_world? (we should actually remove all that from boards, it's just copy&paste every single time)